File: bdd.py
from flask import *

import sqlite3

app = Flask(__name__)

def historial(cur,accion, **kwargs):
	cur.execute("INSERT INTO historial (accion, codigo, idi, descr, lugar) VALUES (?,?,?,?,?)", (accion, 
			kwargs.get("codigo", 0),
			kwargs.get("idi", 0),
			kwargs.get("descr", ""),
			kwargs.get("lugar", 0)))

@app.route("/")
def index():
	return render_template("index.html")

@app.route("/<action>/")
def operacion(action):
	with sqlite3.connect("guardarropia.db") as db:
		db.row_factory = sqlite3.Row
		cur = db.cursor()
		codigo = request.args.get('codigo', None)
		res = {
			"items": [],
			"available": -1,
			"lastinsertion": {},
			"success": False,
			"total" : 0
		}
		actionargs = {}
		try:
			if action == "get":
				cur.execute("select * from guardarropia where vigente = 1 and codigo = ?", (codigo,))
				for row in cur:
					res["items"].append(dict([(x, row[x]) for x in row.keys()]))
			elif action == "add":
				descr = request.args.get("descr", "")
				lugar = request.args.get("lugar", "")
				cur.execute("insert into guardarropia (codigo, descr, lugar, vigente) values (?,?,?, 1)", (codigo, descr, lugar))
				res["lastinsertion"] = {
					"id": cur.lastrowid,
					"codigo": codigo,
					"descr": descr,
					"lugar": lugar
				}
				actionargs = {
					"codigo": codigo,
					"descr": descr,
					"idi": cur.lastrowid,
					"lugar": lugar
				}
			elif action == "mod":
				idi = request.args.get("idi", "")
				descr = request.args.get("descr", "")
				cur.execute("update guardarropia set descr = ? where id = ?", (descr, idi))
				actionargs = {
					"descr": descr,
					"idi": idi
				}
			elif action == "rem": # rem
				idi = request.args.get("idi", "")
				cur.execute("update guardarropia set vigente = 0 where id = ?", (idi,))
				actionargs = {
					"idi": idi
				}
			elif action == "remall": # remall
				cur.execute("update guardarropia set vigente = 0")
			cur.execute("select count(*) from guardarropia where vigente = 1")
			res["total"] = cur.fetchone()[0]
			cur.execute("select lugar from guardarropia where vigente = 1 group by lugar")
			used = set([x[0] for x in cur])
			res["available"] = min(set(range(1,1001)) - used)
			historial(cur, action,**actionargs)
			res["success"] = True
		except Exception:
			res["success"] = False
		return jsonify(res)

if __name__ == "__main__":
	app.run("0.0.0.0", debug=True, port=7000)

File: bdd.php
<?php
	$cx = new SQLite3("guardarropia.db");
	$cx->query("CREATE TABLE IF NOT EXISTS guardarropia (id INTEGER PRIMARY KEY autoincrement, codigo INTEGER, descr TEXT, lugar INTEGER)");
	$action_sql = array(
		"get" => "select * from guardarropia where codigo = %d",
		"add" => "insert into guardarropia (codigo, descr, lugar) values (%d, '%s', %d)",
		"rem" => "delete from guardarropia where id = %d"
	);
	// obtener lugar para el loco, si es que no tiene ya
	$codigo = $_GET['codigo'];

	$response = array(
		"items" => array(),
		"available" => -1,
		"lastinsertion" => 0
		);

	$query = "";
	switch($_GET['action'])
	{
		case "get":
			$query = sprintf($action_sql["get"], $_GET['codigo']);
		break;
		case "add":
			$query = sprintf($action_sql["add"], $_GET['codigo'], $_GET['descr'], $_GET['lugar']);
		break;
		case "rem":
			$query = sprintf($action_sql["rem"], $_GET['idi']);
		break;
	}
	//echo $query;
	$res = $cx->query($query);
	if ($res)
	{
		$response["success"] = true;
		if ($_GET['action'] == "add")
			$response["lastinsertion"] = $cx->lastInsertRowID();
		while($row = $res->fetchArray(SQLITE3_ASSOC))
		{
			$response["items"][] = $row;
		}
		$res2 = $cx->query("select lugar from guardarropia group by lugar");
		$available = range(1,1000);
		while($row = $res2->fetchArray(SQLITE3_ASSOC))
			$available = array_diff($available, array($row['lugar']));
		$response["available"] = min($available);
	}
	else
	{
		$response["success"] = false;
	}
	echo json_encode($response);
?>